src/dune_cache/dune_cache.ml

```diff
@@ -331,9 +331,27 @@ module Cache = struct
 
   let teardown _ = ()
 
-  let detect_duplication_mode _ =
-    (* FIXME: use copy if root is on a different partition *)
-    Duplication_mode.Hardlink
+  let detect_duplication_mode root =
+    if not (Path.Build.build_dir_set ()) then
```

Comment:
This feels a bit hackish. Plus it means re-adding `Fdecl.peek` which was intentionally deleted in #2987. What about:

- making the `duplication_mode` argument of `make` explicit
- explicitly passing `Hardlink` in the daemon
- explicitly calling `detect_duplication_mode` otherwise

?

This way everything is simple and obvious and we don't need to guess anything here.

Reply:
Done ; I'm not a huge fan of a function that crashes the program in the build dir wasn't set though :)

Reply:
You got a point here. Maybe one day we should tackle this issue in a general way.